### PR TITLE
soc/da1469x: Update sys_arch_reboot() function

### PR DESCRIPTION
**Pull request by ydamigos**
Update sys_arch_reboot() function

### CODE REVIEW

File: soc/renesas/smartbond/da1469x/soc.c [hunk 1]
```diff
@@ -36,9 +37,27 @@ static uint32_t z_renesas_cache_configured;
...
+		NVIC_SystemReset();
+	} else if (type == SYS_REBOOT_COLD) {
+		if ((SYS_WDOG->WATCHDOG_REG & SYS_WDOG_WATCHDOG_REG_WDOG_VAL_NEG_Msk) == 0) {
+			/* Cannot write WATCHDOG_REG while WRITE_BUSY */
+			while ((SYS_WDOG->WATCHDOG_REG &
+				SYS_WDOG_WATCHDOG_CTRL_REG_WRITE_BUSY_Msk) != 0) {
+
+			}
+			/* Write WATCHDOG_REG */
+			SYS_WDOG->WATCHDOG_REG = BIT(SYS_WDOG_WATCHDOG_REG_WDOG_VAL_Pos);
```
Comment (ioannis-karachalios) [inline soc/renesas/smartbond/da1469x/soc.c:50]: Updating the counter value should be ignored if `WATCHDOG_REG[WDOG_EN]` is not written to one, at the same time.

File: soc/renesas/smartbond/da1469x/soc.c [hunk 2]
```diff
@@ -36,9 +37,27 @@ static uint32_t z_renesas_cache_configured;
...
+
+			GPREG->RESET_FREEZE_REG = GPREG_SET_FREEZE_REG_FRZ_SYS_WDOG_Msk;
```
Comment (ioannis-karachalios) [inline soc/renesas/smartbond/da1469x/soc.c:52]: Freezing/resuming the watchdog counter requires that `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN]` is first written to one.

**Review by ioannis-karachalios — CHANGES_REQUESTED** [review]

File: soc/renesas/smartbond/da1469x/soc.c (hunk 1 above)
Comment (ydamigos) [reply soc/renesas/smartbond/da1469x/soc.c:50]: image
`WATCHDOG_REG[WDOG_EN]` should be zero to update its value. For sure the register field name is misleading.

File: soc/renesas/smartbond/da1469x/soc.c (hunk 2 above)
Comment (ydamigos) [reply soc/renesas/smartbond/da1469x/soc.c:52]: The goal here is to let the watchdog running and cause a reset.
If `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN] == 0` then the watchdog is running and we are good.
If `WATCHDOG_CTRL_REG[WDOG_FREEZE_EN] == 1` we make sure we unfreeze it.
The same stands for `WATCHDOG_CTRL_REG[NMI_RST]`.

File: soc/renesas/smartbond/da1469x/soc.c (hunk 1 above)
Comment (ioannis-karachalios) [reply soc/renesas/smartbond/da1469x/soc.c:49]: Right. Somehow, I misinterpreted the description.

**Review by ioannis-karachalios — APPROVED** [review]

**Review by blauret — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@ydamigos we are currently merging only bug fixes and documentation updates into the 4.0 release. If this PR is fixing a bug then kindly add a bug report and link it to this PR.

**Comment by ydamigos** [comment]
> @ydamigos we are currently merging only bug fixes and documentation updates into the 4.0 release. If this PR is fixing a bug then kindly add a bug report and link it to this PR.

@mmahadevan108 It adds a cold reboot implementation for SoC.

**Pull request merged**